web/src/main/java/com/softserve/osbb/controller/AddressController.java

```diff
@@ -40,19 +41,33 @@
     @RequestMapping(value = "/city/{id}", method = RequestMethod.GET)
     public ResponseEntity<List<City>> getAllCitiesOfRegion(@PathVariable("id") Integer id) {
         logger.info("Get all cities of region: ");
-        return new ResponseEntity<>(addressService.getAllCitiesOfRegion(id), HttpStatus.OK);
+        if (addressService.getRegionById(id) == null) {
+            return new ResponseEntity<>(new ArrayList<City>(), HttpStatus.NOT_FOUND);
+        } else {
+            return new ResponseEntity<>(addressService.getAllCitiesOfRegion(id), HttpStatus.OK);
```

Comment:
Two return statements - not good.
Please change your code like this:

HttpStatus status = HttpStatus.OK;
List<City> cities = new ArrayList<>();
if (addressService.getRegionById(id) == null) {
            status = HttpStatus.NOT_FOUND;
        } else {
            cities.addAll(addressService.getAllCitiesOfRegion(id));
        }
return new ResponseEntity<>(cities, status);